topKFrequentElements.js

```diff
@@ -0,0 +1,36 @@
+// This file contains solution to the following problem:
+// https://leetcode.com/problems/top-k-frequent-elements/
+
+const topKFrequent = (nums, k) => {
+  const numsOccurences = {};
+  for (let i = 0; i < nums.length; i += 1) {
+    if (!numsOccurences[nums[i]]) {
+      numsOccurences[nums[i]] = 1;
+    } else {
+      numsOccurences[nums[i]] += 1;
+    }
+  }
+  const sortedOccurances = [];
+  const numsOccurencesArr = Object.entries(numsOccurences);
+  for (let i = 0; i < numsOccurencesArr.length; i += 1) {
+    const element = numsOccurencesArr[i];
+    const occurence = element[1];
+    const num = +element[0];
```

Comment:
Need to leave some clarifying comments on what these `occurence` and `num` are (why we use `element[0]` and `element[1]` for those).

Also, there seems to be no need to define it inside the loop ? Move such things outside the loop when possible.

Reply:
I do not think that comments are needed. We count occurences of each number in array and put it in hashmap - numsOccurences. Then we take all the keys and values of hashmap and put them in array like this - [[1, 3], [2, 2], [3, 5]], where occurance of number one is 3, occurence of number 2 is 2, occurence of number 3 is 5.
element[0] is the number
element[1] is the occurence

For loop goes through this array: [[1, 3], [2, 2], [3, 5]]